jlm/hls/backend/rvsdg2rhls/mem-conv.cpp

```diff
@@ -639,16 +639,16 @@ jlm::hls::MemoryConverter(jlm::llvm::RvsdgModule & rm)
       lambda->attributes());
 
   rvsdg::SubstitutionMap smap;
-  for (size_t i = 0; i < lambda->ncvarguments(); ++i)
+  for (const auto & ctxvar : lambda->GetContextVars())
   {
-    smap.insert(
-        lambda->cvargument(i),
-        newLambda->add_ctxvar(lambda->cvargument(i)->input()->origin()));
+    smap.insert(ctxvar.inner, newLambda->AddContextVar(ctxvar.input->origin()).inner);
   }
 
-  for (size_t i = 0; i < lambda->nfctarguments(); ++i)
+  auto args = lambda->GetFunctionArguments();
+  auto newArgs = newLambda->GetFunctionArguments();
+  for (size_t i = 0; i < args.size(); ++i)
```

Comment:
Could we get an assert here: `JLM_ASSERT(args.size() == newArgs.size())`

Reply:
added assert here as well as in the other places